packages/serverless/src/domain/use-case/collection.ts

```diff
@@ -64,14 +80,29 @@ export class Collection {
         );
       }
 
-      // Create the index using ModelCollection
-      const result = ModelCollection.getInstance(
-        databaseName,
-        DATABASE_ENGINE.serverless,
-        collectionName
-      ).index(index, { session });
+      // Perform concurrency task create
+      const indexFillResult = await Fill(
```

Comment:
I think we should rollback on failed, so `Fill` may not work since it won't throw anything.

Reply:
I have session, but I don't sure it can rollback when it's failed or we got race condition, I will test it

Reply:
Fixed, change to for..loop. We can't perform concurrency in transaction